src/libappimage/utils/HashLib.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <istream>
+namespace appimage {
+    namespace utils {
+        /**
+         * C++ wrapper around the bare C hashing algorithms implementations
+         */
+        class HashLib {
```

Comment:
This should not be a class. Just put those methods into a namespace `hashlib` or `hashing` or something.